src/services/NumericalCalculations.jl

```diff
@@ -452,6 +370,43 @@ end
 # sensitive to which hypo of course , see #1024
 #
 
+struct CalcConv end
+struct CalcDeconv end
+
+_getindex_anyn(vec, n) = begin
+  len = length(vec)
+  # 1:len or any random element in that range
+  getindex(vec, n <= len ? n : rand(1:len) )
+end
+
+# NOTE to future self, this will likely become the cost function for Manopt as:
+# function (cf::CalcFactorNormSq)(M::AbstractManifold, x)
+# CalcConv is likeley needed for conv vs deconv
+function (cf::CalcFactorNormSq)(::Type{CalcConv}, x) 
+  
+  sampleIdx = cf._sampleIdx
+  varValsHypo = cf._legacyParams
+  # set the target hypo on the correct sample to free variable x, was target object
+  varValsHypo[cf.solvefor][sampleIdx] = x
+
+  res = cf(cf.measurement[sampleIdx], map(vvh -> _getindex_anyn(vvh, sampleIdx), varValsHypo)...)
+  return sum(x->x^2, res)
+end
+
+function _buildHypoCalcFactor(ccwl::CommonConvWrapper, smpid::Integer)
+  # build a view to the decision variable memory
+  varValsHypo = ccwl.varValsAll[][ccwl.hyporecipe.activehypo]
```

Comment:
Looks good, I just want talk through the one CalcFactor for each sample solve to be sure we're on the right track